components/profile.js

```diff
@@ -32,6 +32,51 @@ SteamCommunity.prototype.setupProfile = function(callback) {
 	});
 };
 
+SteamCommunity.prototype.editShowcaseItem = function(showcase, slot, item, callback) {
+	const self = this;
+	//The possible options, with the maximum number of slots and the corresponding type
+	const allowedoptions = {
+		"trade": {
+			"maxslots": 6,
+			"type": 4
+		},
+		"items": {
+			"maxslots": 10,
+			"type": 3
+		},
+		"games": {
+			"maxslots": 4,
+			"type": 2
+		}
+	};
+
+	if(!allowedoptions.hasOwnProperty(showcase)){
+		const err = new Error("The submitted showcase type has no editable items.");
+		return callback ? callback(err) : undefined;
+	}
+	if(slot < 1 || slot > allowedoptions[showcase]["maxslots"]){
+		const err = new Error("The submitted slot is outside of range. (Allowed range: 1-"+allowedoptions[showcase]["maxslots"]+")");
+		return callback ? callback(err) : undefined;
+	}
+	if(!(item.hasOwnProperty("appid") || item.hasOwnProperty("item_contextid") || item.hasOwnProperty("item_assetid"))){
+		const err = new Error("The submitted item is not valid.");
+		return callback ? callback(err) : undefined;
+	}
+	const requestdata = item;
+	requestdata["slot"] = slot - 1;
+	requestdata["customization_type"] = allowedoptions[showcase]["type"];
+	requestdata["sessionid"] = self.getSessionID();
```

Comment:
ES6!

```JS
const requestData = {
  ...item,
  slot: slot - 1,
  curtomization_type: allowedOptions[showcase].type,
  sessionid = this.getSessionID()
};
```

Besides the look factor, which, let's say, is subjective, this way you do not modify `item` object.

Check this out (minimal reproduction of the code above):

```JS